src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -50,13 +50,20 @@ internal struct Builder
             /// </remarks>
             internal bool InterlockedInitialize(TypeWithAnnotations type)
             {
-                if ((object)_defaultType != null)
+                if (!IsDefault)
                 {
                     return false;
                 }
                 _nullableAnnotation = type.NullableAnnotation;
-                Interlocked.CompareExchange(ref _extensions, type._extensions, null);
-                return (object)Interlocked.CompareExchange(ref _defaultType, type.DefaultType, null) == null;
+                Interlocked.CompareExchange(ref _defaultType, type.DefaultType, null);
+                return Interlocked.CompareExchange(ref _extensions, type._extensions ?? Extensions.Default, null) is null;
```

Comment:
Should we consider asserting that if this returns false that the results were the same?